arduino/smartcar/smartcar.ino

```diff
@@ -78,36 +76,55 @@ void setup() {
  
   mqtt.subscribe("/smartcar/control/#", 1);
   mqtt.onMessage([](String topic, String message) {
-    if (topic == "/smartcar/control/throttle") {
-      car.setSpeed(message.toInt());
-    } else if (topic == "/smartcar/control/steering") {
-      car.setAngle(message.toInt());
-    } else if (topic== "/smartcar/speedometer"){
-            velocity = message.toInt();
-    else {
+    if (topic == "/smartcar/control/takeInput") {
+      takeInput(message);
+    } else {
       Serial.println(topic + " " + message);
     }
   });
 }
  
-void obstacle() {
-    const auto distance = front.getDistance();
- if (distance > 0 && distance < 100) {
-    car.setSpeed(0);
- }
-}
-  //  const auto speed = String(car.getSpeed());
-void carGo() {
-      car.setSpeed(60);
-}
- 
+void takeInput(String input) {
+        int inputSelection = input.substring(0,1).toInt();
+            int appInput;
+            if(input.length() > 1) {
+              unsigned int stringInput = input.substring(1).toInt(); 
+              appInput = stringInput;
+            }
+            
+            switch(inputSelection) {
+              case 2:  //forward
+                car.setSpeed(appInput); // incrementing number from app to go forward
+                break;
+              
+              case 3:  //backwards
+                car.setSpeed(-appInput); // incrementing number from app to go backwards
+                break;
+            
+              case 4:  //right
+                  car.setAngle(-appInput); // incrementing number from app to turn right
+                break;
+            
+              case 5:  //left
+                  car.setAngle(appInput); // incrementing number from app to turn right
+                break;
+                
+              case 7: //stop
+                car.setSpeed(0);
+                car.setAngle(0);
+              break;
+              
+              default:
+                break;
+            }
+        }
+
 void loop() {
  
-  obstacle();
- 
   if (mqtt.connected()) {
     mqtt.loop();
     const auto currentTime = millis();
+    mqtt.publish(speedometer, String(car.getSpeed()));
```

Comment:
I think this line should be removed. This functionality is implemented on line 141.